internal/driver/fetch_test.go

```diff
@@ -206,6 +206,138 @@ func TestFetch(t *testing.T) {
 	}
 }
 
+func TestFetchWithBase(t *testing.T) {
+	const path = "testdata/"
+	// set up testcases
+	type testcase struct {
+		sources         []string
+		bases           []string
+		normalize       bool
+		testDescription string
+		expectSamples   bool
+	}
+
+	testcases := []testcase{
+		{
+			[]string{
+				path + "cppbench.cpu",
+			},
+			[]string{
+				path + "cppbench.cpu",
+			},
+			false,
+			"not normalized, base is same as non-base",
+			false,
+		},
+		{
+			[]string{
+				path + "cppbench.cpu",
+			},
+			[]string{
+				path + "cppbench.cpu",
+				path + "cppbench.cpu",
+			},
+			false,
+			"not normalized, single non-base, multiple base (all profiles same)",
+			true,
```

Comment:
Instead of just validating whether there are samples or not, would it be better to validate the specific value count for one of functions in the profile? Feel free to decline if it seems redundant / not useful.